### PR TITLE
[TritonGEN] Verify HW restrictions for 2D block operators

### PR DESCRIPTION
**Pull request by whitneywhtsang**
The restrictions for OpenCL C builtins are more restrictive than the HW restrictions. 
When using GenISA, we want to only verify the HW restrictions.

### CODE REVIEW

**Review by chengjunlu — APPROVED** [review]

**Review by Dewei-Wang-sh — APPROVED** [review]

File: third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp [hunk 1]
```diff
@@ -164,7 +164,160 @@ LogicalResult TritonGEN::MatrixDPASOp::verify() {
...
+          "expecting elem_size_in_bits to be 8, 16, 32, or 64");
+    }
+
+    return success();
+  }
+
+  if (op.getTranspose()) {
+    if (op.getVnniTransform())
+      return op.emitOpError(
+          "transpose and vnni_transform are mutually exclusive");
```
Comment (etiotto) [inline third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp:239]: I rather have this early exit (when both transpose and transform are true).

File: third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp [hunk 2]
```diff
@@ -164,7 +164,160 @@ LogicalResult TritonGEN::MatrixDPASOp::verify() {
...
+      break;
+    default:
+      return op.emitOpError("transpose is only supported for 32 and 64 bit "
+                            "elements");
+    }
+
+    return success();
+  }
+
+  assert(op.getVnniTransform() && "Expecting vnni_transform should be true");
```
Comment (etiotto) [inline third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp:277]: assert(op.getVnniTransform() && !op.getTranspose() && ...)

**Review by etiotto — APPROVED** [review]

File: third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp]: done.

File: third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp (hunk 2 above)
Comment (whitneywhtsang) [reply third_party/intel/lib/Dialect/TritonGEN/IR/TritonGENOps.cpp]: done.

**Pull request merged**